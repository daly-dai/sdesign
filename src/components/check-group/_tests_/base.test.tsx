import { fireEvent, render } from '@testing-library/react';
import { Checkbox } from 'antd';
import React from 'react';
import { describe, expect, it, vi } from 'vitest';

import SConfigProvider from '../../config-provider';
import SCheckGroup from '../index';

import mountTest from '@dalydb/sdesign/tests/mountTest';

const globalDict = {
  level: {
    L1: '不敏感',
    L2: '低敏感',
    L3: '较敏感',
    L4: '敏感',
    L5: '极敏感',
  },
  type: {
    app_link: '存量应用关联',
    register: '发布注册',
  },
};

describe('checkBoxGroup', () => {
  mountTest(SCheckGroup);

  mountTest(() => <SCheckGroup />);

  it('should render correctly', () => {
    const { container } = render(<SCheckGroup />);
    expect(container).toMatchSnapshot();
  });

  it('should support dictKey', () => {
    const { container } = render(
      <SConfigProvider globalDict={globalDict}>
        <SCheckGroup dictKey="level" />
      </SConfigProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it('should options props > dict props', () => {
    const onChange = vi.fn();

    const { container } = render(
      <SCheckGroup
        dict={globalDict.level}
        onChange={onChange}
        options={[
          {
            label: '存量应用关联',
            value: 'app_link',
          },
          {
            label: '发布注册',
            value: 'register',
          },
        ]}
      />,
    );

    fireEvent.click(container.querySelectorAll('.ant-checkbox-input')[0]);
    expect(onChange).toHaveBeenCalledWith('app_link');
  });

  it('should work basically', () => {
    const onChange = vi.fn();

    const { container } = render(
      <SCheckGroup
        dict={globalDict.level}
        onChange={onChange}
        dictKey="level"
      />,
    );

    fireEvent.click(container.querySelectorAll('.ant-checkbox-input')[0]);
    expect(onChange).toHaveBeenCalledWith('L1');
    fireEvent.click(container.querySelectorAll('.ant-checkbox-input')[1]);
    expect(onChange).toHaveBeenCalledWith('L1,L2');
    fireEvent.click(container.querySelectorAll('.ant-checkbox-input')[2]);
    expect(onChange).toHaveBeenCalledWith('L1,L2,L3');
    fireEvent.click(container.querySelectorAll('.ant-checkbox-input')[1]);
    expect(onChange).toHaveBeenCalledWith('L1,L3');
  });

  it('should support dict', () => {
    const { container } = render(<SCheckGroup dict={globalDict.level} />);

    expect(container).toMatchSnapshot();
  });

  it('should support disableKeys', () => {
    const onChangeGroup = vi.fn();

    const { container } = render(
      <SConfigProvider globalDict={globalDict}>
        <SCheckGroup dictKey="level" disableKeys={['L1']} />
      </SConfigProvider>,
    );

    fireEvent.click(container.querySelectorAll('.ant-checkbox-input')[0]);
    expect(onChangeGroup).not.toHaveBeenCalled();
  });

  it('should support children', () => {
    const onChange = vi.fn();

    const { container } = render(
      <SCheckGroup onChange={onChange}>
        <Checkbox value={'L1'} />
        <Checkbox value={'L2'} />
      </SCheckGroup>,
    );

    fireEvent.click(container.querySelectorAll('.ant-checkbox-input')[0]);
    expect(onChange).toHaveBeenCalledWith('L1');
    fireEvent.click(container.querySelectorAll('.ant-checkbox-input')[1]);
    expect(onChange).toHaveBeenCalledWith('L1,L2');
  });
});
