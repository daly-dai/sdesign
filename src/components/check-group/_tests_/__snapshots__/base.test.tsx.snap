// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`checkBoxGroup > should render correctly 1`] = `
<div>
  <div
    class="ant-checkbox-group css-dev-only-do-not-override-3rel02"
  />
</div>
`;

exports[`checkBoxGroup > should support dict 1`] = `
<div>
  <div
    class="ant-checkbox-group css-dev-only-do-not-override-3rel02"
  >
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-3rel02"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-3rel02"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L1"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        不敏感
      </span>
    </label>
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-3rel02"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-3rel02"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L2"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        低敏感
      </span>
    </label>
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-3rel02"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-3rel02"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L3"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        较敏感
      </span>
    </label>
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-3rel02"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-3rel02"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L4"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        敏感
      </span>
    </label>
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-3rel02"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-3rel02"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L5"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        极敏感
      </span>
    </label>
  </div>
</div>
`;

exports[`checkBoxGroup > should support dictKey 1`] = `
<div>
  <div
    class="ant-checkbox-group css-dev-only-do-not-override-3rel02"
  >
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-3rel02"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-3rel02"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L1"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        不敏感
      </span>
    </label>
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-3rel02"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-3rel02"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L2"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        低敏感
      </span>
    </label>
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-3rel02"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-3rel02"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L3"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        较敏感
      </span>
    </label>
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-3rel02"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-3rel02"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L4"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        敏感
      </span>
    </label>
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-3rel02"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-3rel02"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L5"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        极敏感
      </span>
    </label>
  </div>
</div>
`;

exports[`checkBoxGroup should render correctly 1`] = `
<div>
  <div
    class="ant-checkbox-group css-dev-only-do-not-override-1r287do"
  />
</div>
`;

exports[`checkBoxGroup should support children 1`] = `
<div>
  <div
    class="ant-checkbox-group css-dev-only-do-not-override-1r287do"
  >
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-1r287do"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-1r287do"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L1"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        不敏感
      </span>
    </label>
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-1r287do"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-1r287do"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L2"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        低敏感
      </span>
    </label>
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-1r287do"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-1r287do"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L3"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        较敏感
      </span>
    </label>
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-1r287do"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-1r287do"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L4"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        敏感
      </span>
    </label>
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-1r287do"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-1r287do"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L5"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        极敏感
      </span>
    </label>
  </div>
</div>
`;

exports[`checkBoxGroup should support dict 1`] = `
<div>
  <div
    class="ant-checkbox-group css-dev-only-do-not-override-1r287do"
  >
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-1r287do"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-1r287do"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L1"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        不敏感
      </span>
    </label>
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-1r287do"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-1r287do"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L2"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        低敏感
      </span>
    </label>
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-1r287do"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-1r287do"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L3"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        较敏感
      </span>
    </label>
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-1r287do"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-1r287do"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L4"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        敏感
      </span>
    </label>
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-1r287do"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-1r287do"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L5"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        极敏感
      </span>
    </label>
  </div>
</div>
`;

exports[`checkBoxGroup should support dictKey 1`] = `
<div>
  <div
    class="ant-checkbox-group css-dev-only-do-not-override-1r287do"
  >
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-1r287do"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-1r287do"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L1"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        不敏感
      </span>
    </label>
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-1r287do"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-1r287do"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L2"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        低敏感
      </span>
    </label>
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-1r287do"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-1r287do"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L3"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        较敏感
      </span>
    </label>
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-1r287do"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-1r287do"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L4"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        敏感
      </span>
    </label>
    <label
      class="ant-checkbox-wrapper ant-checkbox-group-item css-dev-only-do-not-override-1r287do"
    >
      <span
        class="ant-checkbox ant-wave-target css-dev-only-do-not-override-1r287do"
      >
        <input
          class="ant-checkbox-input"
          type="checkbox"
          value="L5"
        />
        <span
          class="ant-checkbox-inner"
        />
      </span>
      <span>
        极敏感
      </span>
    </label>
  </div>
</div>
`;
